lib/regexp/regexp.go

```diff
@@ -0,0 +1,290 @@
+	var src string
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &src); err != nil {
+		return nil, err
+	}
+
+	re := b.Receiver().(*Regexp).re
+	return starlark.String(re.FindString(src)), nil
+}
+
+func findAll(_ *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		src string
+		max int = -1
+	)
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &src, &max); err != nil {
+		return nil, err
+	}
+
+	re := b.Receiver().(*Regexp).re
+	return toList(re.FindAllString(src, max)), nil
+}
+
+func findSubmatches(_ *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var src string
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &src); err != nil {
+		return nil, err
+	}
+
+	re := b.Receiver().(*Regexp).re
+	return toList(re.FindStringSubmatch(src)), nil
+}
+
+func replaceAll(thread *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		src  string
+		repl starlark.Value
+	)
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 2, &src, &repl); err != nil {
+		return nil, err
+	}
+
+	re := b.Receiver().(*Regexp).re
+	switch x := repl.(type) {
+	case starlark.Callable:
+		var fnErr error
+		fn := func(matched string) string {
+			res, err := starlark.Call(thread, repl, starlark.Tuple{starlark.String(matched)}, nil)
+			if err != nil {
+				// Save the error to be able to return it to the caller
+				fnErr = fmt.Errorf("%s: An error occured while executing the callable: %s", x.Name(), err.Error())
+				return ""
+			}
+			resp, ok := res.(starlark.String)
+			if !ok {
+				// Save the error to be able to return it to the caller
+				fnErr = fmt.Errorf("%s: A string is expected as return type of the callable but was %s", x.Name(), res.Type())
+				return ""
+			}
+			return string(resp)
+		}
+		result := re.ReplaceAllStringFunc(src, fn)
+		if fnErr != nil {
+			return nil, fnErr
+		}
+		return starlark.String(result), nil
+	case starlark.String:
+		return starlark.String(re.ReplaceAllString(src, convertReplacementPattern(string(x)))), nil
+	}
+	return nil, fmt.Errorf("got %s, want a string or callable", repl.Type())
+}
+
+func split(_ *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		src string
+		max int = -1
+	)
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &src, &max); err != nil {
+		return nil, err
+	}
+
+	re := b.Receiver().(*Regexp).re
+	return toList(re.Split(src, max)), nil
+}
+
+// Converts the replacement pattern to make it compatible with what the regexp package expects.
+func convertReplacementPattern(repl string) string {
+	// Escapes the specifal dollar characters if any
+	repl = strings.ReplaceAll(repl, "$", "$$")
```

Comment:
This will spurious replace an escaped dollar symbol.

Reply:
This is needed to escape existing dollar characters since the dollar character is used to compose a backreference